llvm/tools/opt/CMakeLists.txt

```diff
@@ -29,12 +29,24 @@ set(LLVM_LINK_COMPONENTS
   Passes
   )
 
-add_llvm_tool(opt
+# We don't want to link this into libLLVM
+add_llvm_library(LLVMOptDriver
+  STATIC
```

Comment:
The usual llvlm style is to have one target per directory. If this is now a lib, could you move this into llvm/tools/opt/lib? Then you don't need the  PARTIAL_SOURCES_INTENDED below.